### PR TITLE
Move pkg/templates away

### PR DESCRIPTION
**Pull request by vdemeester**
- Remove unused function and variables from the package
- Remove usage of it from `profiles/apparmor` where it wasn't required
- Move the package to `daemon/logger/templates` where it's only used

Related to https://github.com/docker/cli/pull/427 and https://github.com/moby/moby/issues/32989

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Review by dnephin — APPROVED** [review]
LGTM

**Review by yongtang — APPROVED** [review]
LGTM

**Comment by vieux** [comment]
we went from 4 tests to 1 test, expected ?

File: daemon/logger/templates/templates.go [hunk 1]
```diff
@@ -27,34 +27,6 @@ var basicFunctions = template.FuncMap{
 	"truncate": truncateWithLength,
 }
 
-// HeaderFunctions are used to created headers of a table.
-// This is a replacement of basicFunctions for header generation
-// because we want the header to remain intact.
-// Some functions like `split` are irrelevant so not added.
-var HeaderFunctions = template.FuncMap{
```
Comment (stevvooe) [inline daemon/logger/templates/templates.go]: This was odd. Do we need to maintain this behavior?

**Review by stevvooe — APPROVED** [review]
LGTM

Looks like we've removed some dead code but wanted to confirm there wasn't functionality we are removing.

File: daemon/logger/templates/templates.go (hunk 1 above)
Comment (vdemeester) [inline daemon/logger/templates/templates.go]: @stevvooe @vieux yes, this part is only used `cli` side, so I moved it there 👼

**Comment by tiborvass** [comment]
LGTM

Ping @tophj-ibm for z failure: https://jenkins.dockerproject.org/job/Docker-PRs-s390x/4934/console

**Comment by tophj-ibm** [comment]
@tiborvass thanks, I'm not sure why z is timing out everywhere, but I let that team know. I also increased the timeout for z temporarily from 180 -> 300m, so hopefully it doesn't fail in the meantime.

**Comment by thaJeztah** [comment]
Thanks @tophj-ibm !

**Pull request merged**